src/amd/compiler/aco_live_var_analysis.cpp

```diff
@@ -209,31 +188,29 @@ void process_live_temps_per_block(Program *program, live& lives, Block* block,
       abort();
    }
 
-   assert(block->index != 0 || (vgpr_demand == 0 && (sgpr_demand == 0)));
+   assert(block->index != 0 || (new_demand.vgpr == 0 && (new_demand.sgpr == 0)));
```

Comment:
That we used unnecessary braces doesn't mean you are allowed to do this as well :P
maybe just assert(block->index != 0 || new_demand == RegisterDemand(0, 0))